databrowser/src/components/language/LanguagePicker.vue

```diff
@@ -63,45 +64,55 @@ export default defineComponent({
     PillGroup,
     ArrowDown,
   },
-  setup() {
-    const languageParameter = useUrlQueryParameter('language', 'en', {
-      defaultValue: 'en',
+  props: {
+    defaultLanguage: {
+      type: String as PropType<FilterLanguage>,
+      default: FilterLanguage.EN,
+    },
+  },
+  setup(props) {
+    const supportedLanguages: Array<string> = Object.values(FilterLanguage);
+    const showMobileSelect = ref<boolean>(false);
+    const languageParameter = useUrlQueryParameter(
+      'language',
+      props.defaultLanguage,
+      {
+        defaultValue: props.defaultLanguage,
+      }
+    );
+
+    const currentSelected = computed(() => {
+      if (!languageParameter.value) {
+        return props.defaultLanguage;
+      }
+
+      return supportedLanguages.includes(languageParameter.value)
+        ? languageParameter.value
+        : props.defaultLanguage;
     });
+
+    function closeDialog() {
+      showMobileSelect.value = false;
+    }
+
+    function changeLanguage(language: string) {
+      languageParameter.value = language;
+      closeDialog();
+    }
+
+    function isSelected(current: string) {
+      return currentSelected.value == current;
+    }
```

Comment:
Here you are using this method to determine if a language is selected. The problem here with Vue is that there does not really exist a strategy to calculate the state of every single language with a computed property.

A possible solution is just to extract the language into a dedicated component where you can caluclate the selected state with an computed property. This is a performance boost. That's why I would extract this into a dedicated component.